addons/portal/static/src/xml/portal_chatter.xml

```diff
@@ -78,38 +100,52 @@
+                        <img class="o_portal_chatter_avatar" t-att-src="message.author_avatar_url" alt="avatar"/>
+                        <div class="media-body">
+                            <div class="o_portal_chatter_message_title row mb-1 mx-0">
+                                <h5 class='mb-0'><t t-esc="message.author_id[1]"/></h5>
+                                <p class="mx-2 pt-1">-</p>
+                                <p class="o_portal_chatter_published_date pt-1" t-att-title="message.published_date_str"><t t-esc="message.date_ago"/></p>
+                                <t t-call="portal.chatter_internal_toggle" t-if="widget.options['is_user_employee']"/>
+                            </div>
+                            <t t-out="message.body"/>
+                            <div t-if="message.attachment_ids.length" class="o_portal_chatter_attachments">
+                                <t t-call="portal.chatter_attachments_link">
+                                    <t t-set="attachments" t-value="message.attachment_ids"/>
+                                    <t t-set="attachment_count" t-value="message.attachment_ids.length"/>
+                                </t>
+                            </div>
                         </div>
                     </div>
-                </div>
+                </t>
             </t>
         </div>
     </t>
 
     <!-- Chatter: internal toggle widget -->
     <t t-name="portal.chatter_internal_toggle">
-        <div t-attf-class="float-right o_portal_chatter_js_is_internal #{message.is_internal and 'o_portal_message_internal_on' or 'o_portal_message_internal_off'}"
+        <div t-attf-class="float-right o_boolean_toggle custom-control custom-checkbox ml-auto #{!message.internal ? 'o_portal_chatter_js_is_internal' : ''}"
                 t-att-data-message-id="message.id"
                 t-att-data-is-internal="message.is_internal">
-            <button class="btn btn-danger"
-                title="Currently restricted to internal employees, click to make it available to everyone viewing this document.">Employees Only</button>
-            <button class="btn btn-success"
-                title="Currently available to everyone viewing this document, click to restrict to internal employees.">Visible</button>
+            <input type="checkbox" t-att-id="'message_' + message.id" class="custom-control-input"
+                t-att-checked="(!message.is_internal and !message.internal and 'checked') or undefined"
+                t-att-disabled="message.internal or undefined"/>
+            <label t-att-for="'message_' + message.id" class="custom-control-label">
+                &#8203;
+                <i t-att-class="'fa fa-times-circle'"
+                    t-att-title="message.internal and 'Log note, only available to internal employees.'
+                                 or 'Currently restricted to internal employees, click to make it available to everyone viewing this document.'"></i>
```

Comment:
This is falsy as ACLs prevent from reading message linked to internal subtype to portal/public. If this works, there is a bug ^^ .

Reply:
Is it a problem if an internal user visiting the portal sees internal subtype messages ? This currently works.